### PR TITLE
Add slack notifications

### PR DESCRIPTION
**Pull request by HotThoughts**
**Proposed changes**:
- Add slack notifications of release using [action-slack](https://github.com/Ilshidur/action-slack)
- Add a testing webhook into secret as `SLACK_WEBHOOK`

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa-sdk/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa-sdk#code-style) for instructions)

### CODE REVIEW

File: .github/workflows/continous-integration.yml [hunk 1]
```diff
@@ -162,15 +162,33 @@ jobs:
...
+    - name: Notify Slack & Publish Release Notes 🗞
+      uses: Ilshidur/action-slack@2.1.0
+
       env:
+        RASA_SDK_VERSION: ${{ github.ref }}
+        SLACK_WEBHOOK: ${{ SLACK_WEBHOOK }}
         GITHUB_TAG: ${{ github.ref }}
         GITHUB_REPO_SLUG: ${{ github.repository }}
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       run: |
```
Comment (tczekajlo) [inline .github/workflows/continous-integration.yml:174]: You can't use `uses` together with `run`. In this case, where you use a GH Action, we don't need to use the `run` argument.

File: .github/workflows/continous-integration.yml [hunk 2]
```diff
@@ -162,15 +162,33 @@ jobs:
         user: __token__
         password: ${{ secrets.PYPI_TOKEN }}
 
-    - name: Publish Release Notes 🗞
-      if: env.IS_TAG_BUILD
+    - name: Notify Slack & Publish Release Notes 🗞
```
Comment (tczekajlo) [inline .github/workflows/continous-integration.yml:165]: We should leave the `Publish Release Notes` step as it is and add new steps for sending Slack notifications that depend on the `Publish Release Notes` step.

File: .github/workflows/continous-integration.yml [hunk 3]
```diff
@@ -162,15 +162,33 @@ jobs:
...
+      with:
+        args: "⚡ New *Rasa SDK* version ${RASA_SDK_VERSION} has been released! Changelog: https://github.com/RasaHQ/rasa-sdk/blob/${RASA_SDK_VERSION}/CHANGELOG.mdx"
+
+    - name: Notify Slack of failure ⛔️
+      uses: Ilshidur/action-slack@2.1.0
+      env:
+        RASA_SDK_VERSION: ${{ github.ref }}
+        SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
+      run: |
+        RASA_SDK_VERSION=${RASA_SDK_VERSION/refs\/tags\//}
```
Comment (tczekajlo) [inline .github/workflows/continous-integration.yml:191]:
```suggestion
```

**Review by tczekajlo — CHANGES_REQUESTED** [review]
@HotThoughts Congrats on your first PR 🎉  I've added few comments

File: .github/workflows/continous-integration.yml (hunk 1 above)
Comment (HotThoughts) [reply .github/workflows/continous-integration.yml:174]: Thank you. 👍🏻 That's the reason why the CI workflow didn't even start.

**Comment by sara-tagger** [comment]
Thanks for submitting a pull request 🚀 @m-vdb will take a look at it as soon as possible ✨

**Comment by sara-tagger** [comment]
Thanks for submitting a pull request 🚀 @b-quachtran will take a look at it as soon as possible ✨

File: .github/workflows/continous-integration.yml [hunk 4]
```diff
@@ -187,5 +195,6 @@ jobs:
       env:
         SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
       uses: Ilshidur/action-slack@2.1.0
+      if: success()
       with:
         args: "⚡ New *Rasa SDK* version {{ github.event.release.tag_name }} has been released! Changelog: https://github.com/RasaHQ/rasa-sdk/blob/{{ github.event.release.tag_name }}/CHANGELOG.mdx"
```
Comment (tczekajlo) [inline .github/workflows/continous-integration.yml]: The `github.event.release.tag_name` field is only available for a `release` event. We don't use this kind of event for the CI workflow.

I'd create an additional step that publishes as output a tag name and then you can use it in the slack messeges.

**Review by tczekajlo — CHANGES_REQUESTED** [review]

File: .github/workflows/continous-integration.yml [hunk 5]
```diff
@@ -88,89 +88,119 @@ jobs:
...
+    if: always()
+    needs:
+      - deploy
+
+    steps:
+      - name: Get RASA SDK Version
+        env:
+          RASA_SDK_VERSION: ${{ github.ref }}
+        run: |
+          RASA_SDK_VERSION=${RASA_SDK_VERSION/refs\/tags\//}
```
Comment (tczekajlo) [inline .github/workflows/continous-integration.yml:190]: You have to set this variable as an environment variable.

https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable
```suggestion
          echo "RASA_SDK_VERSION=${RASA_SDK_VERSION/refs\/tags\//}" >> $GITHUB_ENV
```

File: .github/workflows/continous-integration.yml [hunk 6]
```diff
@@ -88,89 +88,119 @@ jobs:
...
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Read Poetry Version 🔢
+        run: |
+          echo "POETRY_VERSION=$(scripts/poetry-version.sh)" >> $GITHUB_ENV
+        shell: bash
+
+      - name: Install poetry 🦄
```
Comment (tczekajlo) [inline .github/workflows/continous-integration.yml:104]: What happened here? Did your IDE change formatting?

**Review by tczekajlo — CHANGES_REQUESTED** [review]

File: .github/workflows/continous-integration.yml (hunk 6 above)
Comment (HotThoughts) [reply .github/workflows/continous-integration.yml:104]: Didn't realize my IDE formatted it😱

Comment (HotThoughts) [reply .github/workflows/continous-integration.yml]: Reverted to previous commit.

**Review by tczekajlo — APPROVED** [review]
LGTM

**Pull request merged**